pkg/services/promtypemigration/prom_mig.go

```diff
@@ -0,0 +1,89 @@
+package promtypemigration
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/repo"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type PromMigrationHandler interface {
+	Migrate(context.Context, *promMigrationService) error
+	getPrometheusDataSources(ctx context.Context, dataSourcesService datasources.DataSourceService) ([]*datasources.DataSource, error)
+}
+
+type dataSourceService interface {
+	DecryptedValues(ctx context.Context, ds *datasources.DataSource) (map[string]string, error)
+	UpdateDataSource(ctx context.Context, cmd *datasources.UpdateDataSourceCommand) (*datasources.DataSource, error)
+	GetDataSourcesByType(ctx context.Context, query *datasources.GetDataSourcesByTypeQuery) ([]*datasources.DataSource, error)
+}
+
+type promMigrationService struct {
+	cfg                *setting.Cfg
+	dataSourcesService dataSourceService
+	pluginStore        pluginstore.Store
```

Comment:
nit: you can use [pluginRegistry](https://github.com/grafana/grafana/blob/andrew/prom-type-migration-service/pkg/services/pluginsintegration/pluginstore/store.go#L28) which is what pluginStore wraps.